### ISSUE TITLE
Types error in strict: "Cannot invoke an object which is possibly 'undefined'"

### ISSUE THREAD

**Issue by hirvesh** [LegendApp/legend-state#13]
Hello, I am trying to assign a value as per examples in docs as so:

```
const obs = observable({ text: "hi" });
// Assign
obs.assign({ text: "ok" });
```

However, TypeScript is throwing the error:

`Cannot invoke an object which is possibly 'undefined'.`

<img width="542" alt="CleanShot 2022-10-02 at 19 05 31@2x" src="https://user-images.githubusercontent.com/67500738/193461228-c346d6fd-dc25-44f7-91d5-28834639420c.png">

Not sure what is wrong here 🤔

**Comment by jmeistrich** [comment]
This is an issue with `strict: true` in tsconfig. We will look at improving the types to work better with strict mode.

For now you should be able to fix this with:
`obs.assign?.({ text: "ok" });`

**Comment by hirvesh** [comment]
Hey thanks - that worked!

However, I am facing another issue when importing and running the following on Expo:

```
import { enableLegendStateReact } from "@legendapp/state/react";

enableLegendStateReact();
```

 
![CleanShot 2022-10-02 at 20 02 34](https://user-images.githubusercontent.com/67500738/193463885-95ba9bfd-8dcf-4931-b8c8-699346f2a0f5.png)

**Comment by jmeistrich** [comment]
It seems to have to do with us exporting .mjs files. I'm not sure what the correct solution is, will do some more research. If you google for: "can't import the named export from non ecmascript module" there's a bunch of solutions but I'm not sure which would solve this problem without a test repo. Can you try some of those solutions or share a repo so I can try it?

**Comment by ckruppe** [comment]
We have the same problem at work. For us strict mode is mandatory. And we think its not feasable to write always "get?.()". We looked in the typings a bit and the question arises: Why are get, set, assign and peek optional? Does it have to do with library internals?

**Comment by hadnet** [comment]
Hey, @jmeistrich I got same error type as @hirvesh, no types definitions at all when importing components from @legendapp/state/react, any solution on this?

**Comment by jmeistrich** [comment]
@hadnet I haven't had a chance to look into it yet. If you have time to look into it can you try some of the solutions found by googling the error? That would help a lot with being able to track down the problem.

> It seems to have to do with us exporting .mjs files. I'm not sure what the correct solution is, will do some more research. If you google for: "can't import the named export from non ecmascript module" there's a bunch of solutions but I'm not sure which would solve this problem without a test repo. Can you try some of those solutions or share a repo so I can try it?

Otherwise I will try to look into it soon.

**Comment by jmeistrich** [comment]
This seems to be a problem with Expo Web not supporting .mjs files. It can be fixed with this:

Add a `webpack.config.js` file at the root of the Expo project containing this:

```js
const createExpoWebpackConfigAsync = require('@expo/webpack-config');

module.exports = async function (env, argv) {
  const config = await createExpoWebpackConfigAsync(env, argv);
  config.module.rules.push(
    {
      test: /\.mjs$/,
      include: /node_modules/,
      type: "javascript/auto"
  });

  return config;
};
```

I'll update the docs to include something about this, and look into changing the exports in a future version to hopefully not need this.

**Comment by jmeistrich** [comment]
This types error should be fixed in `0.21.18`. Please let me know if it's still not working.
